core/src/Streamly/Internal/Data/Unbox.hs

```diff
@@ -362,48 +362,71 @@ class Unbox a where
     -- | Write an element of type "a" to a MutableByteArray given the byte
     -- index.
     --
-    -- IMPORTANT: The implementation of this interface may not check the bounds
-    -- of the array, the caller must not assume that.
+    -- IMPORTANT: The implementation of this interface does not check the
+    -- bounds of the array.
     pokeByteIndex :: Int -> MutableByteArray -> a -> IO ()
 
     {-# INLINE pokeByteIndex #-}
     default pokeByteIndex :: (Generic a, PokeRep (Rep a)) =>
         Int -> MutableByteArray -> a -> IO ()
     pokeByteIndex i arr = genericPokeByteIndex arr i
 
--- XXX Add asserts to check bounds
+{-# INLINE checkBounds #-}
+checkBounds :: String -> Int -> MutableByteArray -> IO ()
+checkBounds _label _off _arr = do
+#ifdef DEBUG
```

Comment:
Did you check the compilation with this flag enabled?

Reply:
Added a -DDEBUG cpp macro during compliation when debug build flag is on.